include/layers/Tensor.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <stdexcept>
+#include <type_traits>
+#include <vector>
+
+#include "layers/Shape.hpp"
+
+enum class Type { kUnknown, kInt, kDouble };
+
+template <typename T>
+std::vector<uint8_t>* to_byte(std::vector<T>& v) {
+  return reinterpret_cast<std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+const std::vector<uint8_t>* to_byte(const std::vector<T>& v) {
+  return reinterpret_cast<const std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+Type GetTypeEnum() {
+  if constexpr (std::is_same<T, int>::value) {
+    return Type::kInt;
+  } else if constexpr (std::is_same<T, double>::value) {
+    return Type::kDouble;
+  } else {
+    return Type::kUnknown;
+  }
+}
```

Comment:
else is redundant here
```suggestion
  if constexpr (std::is_same<T, int>::value)
    return Type::kInt;
  if constexpr (std::is_same<T, double>::value)
    return Type::kDouble;
  return Type::kUnknown;
}
```

Reply:
Fixed

Reply:
Do I understand correctly that with "else" stmts it doesn't give that?